### ISSUE TITLE
SC1072/SC1073 false positives for valid while loop

### ISSUE THREAD

**Issue by e40** [koalaman/shellcheck#2180]
version: 0.7.1

```sh
    while read line; do
	# this stuff happens in the same shell as the main script
    done <<< "$(cat "$tempfile")"
```
This is a very handy way to iterate over the contents of a temp file, but have the iteration be in the same shell (not a subshell).

```
In template.sh line 85:
    while read line; do
    ^-- SC1073: Couldn't parse this while loop. Fix to allow more checks.
                     ^-- SC1060: Can't have empty do clauses (use 'true' as a no-op).


In template.sh line 87:
    done <<< "$(cat "$tempfile")"
         ^-- SC1072: Unexpected keyword/token. Fix any mentioned problems and try again.
```

**Comment by 0x326** [comment]
The syntax of the snippet is incorrect. It should be:
```bash
    while read line; do
	# this stuff happens in the same shell as the main script
        :
    done <<< "$(cat "$tempfile")"
```
